app.py

```diff
@@ -120,23 +104,27 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
 db_secret_name = database.pgstac.secret.secret_name
 db_security_group = database.db_security_group
 
+base_api_url = f"https://{veda_app_settings.stage_name()}.{veda_app_settings.veda_custom_host}".strip(
+    "/"
+)
+stac_api_url = f"{base_api_url}{veda_app_settings.veda_stac_root_path}/"
+raster_api_url = f"{base_api_url}{veda_app_settings.veda_raster_root_path}/"
+
 # ingestor config requires references to other resources, but can be shared between ingest api and bulk ingestor
 ingestor_config = ingest_config(
     stage=veda_app_settings.stage_name(),
     stac_db_security_group_id=db_security_group.security_group_id,
-    stac_api_url=stac_api.stac_api.url,
-    raster_api_url=raster_api.raster_api.url,
+    stac_api_url=stac_api_url,
```

Comment:
I think we could handle inferring the cloudfront urls in config with a property or function
```
@property
def veda_stac_api_cf_url(self) -> str:
    """inferred cloudfront url of the stac api if app is configured with a custom host and root path"""
    if self.veda_custom_host and self.veda_stac_root_path:
        return f"https://{self.veda_custom_host}{self.veda_stac_root_path}"
    return None 
```

and then conditionally use them here so that we have a way to use the API gateway url if a cloudfront is not used. 

```
stac_api_url=settings.stac_api_cf_url if settings.stac_api_cf_url else stac_api.stac_api.url
```